src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarilyTests.cs

```diff
@@ -0,0 +1,169 @@
+    protected internal static sbyte SByteValueStatic = -1;
+    protected static int Int16ValueStatic = -1;
+    private protected static uint UInt16ValueStatic = 1;
+    public static int Int32ValueStatic = -1;
+    internal static uint UInt32ValueStatic = 1;
+    public static long Int64ValueStatic = long.MaxValue;
+    public static ulong UInt64ValueStatic = ulong.MaxValue;
+    static float s_someFloatStatic = 1.0f;
+    static readonly double s_someDoubleStatic = 1.0;
+    public static System.DateTime CurrentTimeStatic = System.DateTime.Now;
+
+    private object SyncObjInstance = new object();
+    private int? NullableIntInstance = 0;
+    private string EmptyStringValueInstance = """";
+    private string StringValueInstance = ""hello"";
+    private char charValueInstance= '\r';
+    private bool _boolValueInstance = true;
+    public byte ByteValueInstance = 1;
+    protected internal sbyte SByteValueInstance = -1;
+    protected int Int16ValueInstance = -1;
+    private protected uint UInt16ValueInstance = 1;
+    public int Int32ValueInstance = -1;
+    internal uint UInt32ValueInstance = 1;
+    public long Int64ValueInstance = long.MaxValue;
+    public ulong UInt64ValueInstance = ulong.MaxValue;
+    float _someFloatInstance = 1.0f;
+    readonly double _someDoubleInstance = 1.0;
+    public System.DateTime CurrentTimeInstance = System.DateTime.Now;
+
+    public System.DayOfWeek Week1 = (System.DayOfWeek)1;
+    public System.DayOfWeek Week2 = System.DayOfWeek.Sunday;
+}");
+        }
+
+        [Fact]
+        public async Task NoDiagnostics_Nullable()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+#nullable enable
+public class C
+{
+    private static object s_obj1 = null!;
+}
+public class G<T>
+{
+    private T _value = default!;
+}
+",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp8,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task Diagnostics_InitializerRemoved()
+        {
+            await VerifyCS.VerifyCodeFixAsync(
+@"public class C
+{
+    private static object SyncObjStatic [|= null|];
+    private static int? NullableIntStatic [|= null|];
+    private static string StringValueStatic [|= null|];
+    private static char charValueStatic[|= '\0'|];
+    private static bool s_boolValueStatic [|= false|];
+    public static byte ByteValueStatic [|= 0|];
+    protected internal static sbyte SByteValueStatic     [|= 0|];
+    protected static int Int16ValueStatic [|= 0|];
+    private protected static uint UInt16ValueStatic [|= 0|];
+    public static int Int32ValueStatic [|= 0|];
+    internal static uint UInt32ValueStatic [|= 0|];
+    public static long Int64ValueStatic [|= 0|];
+    public static ulong UInt64ValueStatic [|= 0|];
+    static float s_someFloatStatic [|= 0f|];
+    static readonly double s_someDoubleStatic [|= 0|];
+    public static System.DateTime CurrentTimeStatic [|= default|];
```

Comment:
Add tests for:

1. Declarations with multiple fields, where:
    1. Both have required initializers
    2. One of them has required initializer
    3. Neither have required initializers
    4. Fields with shared initializer (VB As New clause seems to be the only one I think)

2. Simple VB test for sanity